ev3dev/core.py

```diff
@@ -2787,33 +2789,49 @@ class InfraredSensor(Sensor, ButtonBase):
     # pressed.
     _BUTTON_VALUES = {
             0: [],
-            1: ['red_up'],
-            2: ['red_down'],
-            3: ['blue_up'],
-            4: ['blue_down'],
-            5: ['red_up', 'blue_up'],
-            6: ['red_up', 'blue_down'],
-            7: ['red_down', 'blue_up'],
-            8: ['red_down', 'blue_down'],
+            1: ['top_left'],
+            2: ['bottom_left'],
+            3: ['top_right'],
+            4: ['bottom_right'],
+            5: ['top_left', 'top_right'],
+            6: ['top_left', 'bottom_right'],
+            7: ['bottom_left', 'top_right'],
+            8: ['bottom_left', 'bottom_right'],
             9: ['beacon'],
-            10: ['red_up', 'red_down'],
-            11: ['blue_up', 'blue_down']
+            10: ['top_left', 'bottom_left'],
+            11: ['top_right', 'bottom_right']
             }
 
-    #: Handles ``Red Up`` events.
-    on_red_up = None
-
-    #: Handles ``Red Down`` events.
-    on_red_down = None
-
-    #: Handles ``Blue Up`` events.
-    on_blue_up = None
-
-    #: Handles ``Blue Down`` events.
-    on_blue_down = None
-
-    #: Handles ``Beacon`` events.
-    on_beacon = None
+    _BUTTONS = ('top_left', 'bottom_left', 'top_right', 'bottom_right', 'beacon')
+
+    # See process() for an explanation on how to use these
+    #: Handles ``Red Up``, etc events on channel 1
+    channel1_top_left = None
```

Comment:
I just noticed that this drops `on_` prefix from the handlers. I think its ok given that the names differ from `top_left`. `top_right`, ... properties, and are already quite verbose with `channel*_` prefixes. Just wanted to make sure this is intentional. It could be confusing for users (e.g. `channel1_top_left` could be easily misunderstood to be a boolean property for top left button on channel 1).

Reply:
I'll put the `on_` back that way the format is consistent with the `on_left`, `on_right`, etc for the buttons